metis/lib/archiver.rb

```diff
@@ -29,6 +29,14 @@ def archive(data_block)
       data_block.save
     end
 
+    def delete(data_block)
+      raise ArgumentError, "No vault defined!" unless @config
+      raise ArgumentError, "Data block not archived!" unless data_block.archive_id
```

Comment:
Hmm, is this useful?
I think it's actually better to  return false on nil here.  It would allow for easier "sweeps" across datasets without additional logic on the consuming side.
The contract would be "if there is any reason to think this can be deleted from an archive, it will be deleted."